library-client-rtsp/src/main/java/com/alexvas/rtsp/codec/AudioDecodeThread.kt

```diff
@@ -1,171 +1,177 @@
-                    else -> {
-                        if (outIndex >= 0) {
-                            val byteBuffer: ByteBuffer? = decoder.getOutputBuffer(outIndex)
-
-                            val chunk = ByteArray(bufferInfo.size)
-                            byteBuffer?.get(chunk)
-                            byteBuffer?.clear()
-
-                            if (chunk.isNotEmpty()) {
-                                audioTrack.write(chunk, 0, chunk.size)
-                            }
-                            decoder.releaseOutputBuffer(outIndex, false)
-                        }
-                    }
-                }
+                decoder.stop()
+                decoder.release()
+            } catch (e: InterruptedException) {
             } catch (e: Exception) {
                 e.printStackTrace()
             }
-
-            // All decoded frames have been rendered, we can stop playing now
-            if (bufferInfo.flags and MediaCodec.BUFFER_FLAG_END_OF_STREAM != 0) {
-                Log.d(TAG, "OutputBuffer BUFFER_FLAG_END_OF_STREAM")
-                break
-            }
+            audioFrameQueue.clear()
+            if (DEBUG) Log.d(TAG, "$name stopped")
         }
-        audioTrack.flush()
-        audioTrack.release()
 
-        try {
-            decoder.stop()
-            decoder.release()
-        } catch (e: InterruptedException) {
-        } catch (e: Exception) {
-            e.printStackTrace()
-        }
-        audioFrameQueue.clear()
-        if (DEBUG) Log.d(TAG, "$name stopped")
-    }
-
-    companion object {
-        private val TAG: String = AudioDecodeThread::class.java.simpleName
-        private const val DEBUG = false
-
-        fun getAacDecoderConfigData(audioProfile: Int, sampleRate: Int, channels: Int): ByteArray {
-            // AOT_LC = 2
-            // 0001 0000 0000 0000
-            var extraDataAac = audioProfile shl 11
-            // Sample rate
-            when (sampleRate) {
-                7350 -> extraDataAac = extraDataAac or (0xC shl 7)
-                8000 -> extraDataAac = extraDataAac or (0xB shl 7)
-                11025 -> extraDataAac = extraDataAac or (0xA shl 7)
-                12000 -> extraDataAac = extraDataAac or (0x9 shl 7)
-                16000 -> extraDataAac = extraDataAac or (0x8 shl 7)
-                22050 -> extraDataAac = extraDataAac or (0x7 shl 7)
-                24000 -> extraDataAac = extraDataAac or (0x6 shl 7)
-                32000 -> extraDataAac = extraDataAac or (0x5 shl 7)
-                44100 -> extraDataAac = extraDataAac or (0x4 shl 7)
-                48000 -> extraDataAac = extraDataAac or (0x3 shl 7)
-                64000 -> extraDataAac = extraDataAac or (0x2 shl 7)
-                88200 -> extraDataAac = extraDataAac or (0x1 shl 7)
-                96000 -> extraDataAac = extraDataAac or (0x0 shl 7)
+        companion object {
```

Comment:
Why you moved that code? Revert it.

Reply:
That happened during mering, will revert it.